docs/proposals/secret-configmap-file-mode.md

```diff
@@ -0,0 +1,111 @@
+
+ * Change the default file mode for secrets: I think this is unacceptable as it
+   is stated in the api_changes doc. And besides it doesn't feel correct IMHO, it
+   is technically one option. The argument for this might be that world and group
+   readable for a secret is not a nice default, we already take care of not
+   writing it to disk, etc. but the file is created world-readable anyways. Such a
+   default change has been done recently: the default was 0444 in kubernetes <= 1.2
+   and is now 0644 in kubernetes >= 1.3 (and the file is not a regular file,
+   it's a symlink now). This change was done here to minimize differences between
+   configmaps and secrets: https://github.com/kubernetes/kubernetes/pull/25285. But
+   doing it again, and changing to something more restrictive (now is 0644 and it
+   should be 0400 to work with ssh and most apps) seems too risky, it's even more
+   restrictive than in k8s 1.2. Specially if there is no way to revert to the old
+   permissions and some use case is broken by this. And if we are adding a way to
+   change it, like in the option above, there is no need to rush changing the
+   default. So I would discard this.
+
+ * We don't want to people be able to change this, at least for now, and the
+   ones who do, suggest that do it as a "postStart" command. This is acceptable
+   if we don't want to change kubernetes core for some reason, although there
+   seem to be valid use cases. But if the user want's to use the "postStart" for
+   something else, then it is more disturbing to do both things (have a script
+   in the docker image that deals with this, but is not probably concern of the
+   project so it's not nice, or specify several commands by using "sh").
+
+# Proposed implementation
+
+The proposed implementation goes with the first alternative: adding a `mode` to
+the API.
+
+The proposition is to add it as a field in the KeyToPath. That is, as
+v1.SecretVolumeSource in the API[5] and the equivalent for configmaps[4]. And
+then change the key to be optional, but is mandatory to specify at least one of
+`Mode` and `Key`.
+
+Adding it there allows the user to change the mode bits of every file in the
+object, so it achieves the goal.
+
+The are two downside:
+
+ * The files are symlinks pointint to the real file, and the realfile
+   permissions are only set. The symlink has the clasic symlink permissions.
+   This is something already present in 1.3, and it seems applications like ssh
+   work just fine with that. Something worth mentioning, but doesn't seem to be
+   an issue.
+ * If the secret/configMap is mounted in more than one container, the file
+   permissions will be the same on all. This is already the case for Key
+   mappings and doesn't seem like a big issue.
+
+This setting might conflict with the `fsGroup` specified, and in that case if
+the `mode` bits are present for a file the `fsGroup` will be ignored. The reason
```

Comment:
I do think you should spec that we AND the provided mask with 0777 before applying it.  I don't think we should allow users to set sticky and setuid bits, without fully understanding the use case.  We can revisit that later if anyone complains.